src/main/java/com/pubnub/api/models/consumer/objects_vsp/space/Space.java

```diff
@@ -0,0 +1,38 @@
+package com.pubnub.api.models.consumer.objects_vsp.space;
+
+import com.pubnub.api.models.consumer.objects_api.PNObject;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+import lombok.experimental.Accessors;
+
+@Getter
+@Setter
+@Accessors(chain = true)
+@EqualsAndHashCode(callSuper = true)
+@ToString(callSuper = true)
+@NoArgsConstructor
+public class Space extends PNObject {
+    private String name;
+    private String description;
+    private String status;
+    private String type;
+
+    public Space(String id, String name, String description) {
+        super(id);
+        this.name = name;
+        this.description = description;
+    }
+
+    public Space(String id, String name) {
+        this(id, name, null);
+    }
+
+    @Override
+    public PNObject setCustom(Object custom) {
```

Comment:
I think `custom` should not be an `Object` but rather should have the same type as we expect when we `createSpace`, so `Map<String, Object>`. 

What would you say about this simpler version of `Space`?

```
@Data
public class Space {
    @NotNull
    private final SpaceId id;
    private final String name;
    private final String description;
    private final String status;
    private final String type;
    private final Map<String, Object> custom;
}
```

probably we'd need custom deserializer for `SpaceId`

Reply:
I think this is good improvement. I guess we need to add `eTag` and `updated` field so:

@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Data
public class Space extends ObjectVsp {

    @EqualsAndHashCode.Include
    @JsonAdapter(SpaceIdDeserializer.class)
    @Setter
    private SpaceId id;
    private String name;
    private String description;
    private String status;
    private String type;

    @Override
    public Space setCustom(Map<String, Object> custom) {
        super.setCustom(custom);
        return this;
    }
}

@Data
@Accessors(chain = true)
public class ObjectVsp {

    @JsonAdapter(CustomPayloadJsonInterceptor.class)
    @Setter
    protected Map<String, Object> custom;

    protected String updated;
    protected String eTag;
}